src/test/java/com/karankumar/bookproject/ui/book/BookFormTest.java

```diff
@@ -479,14 +517,169 @@ void shouldDisplaySeriesPosition_withSeriesPositionPopulated_whenBookHasSeriesPo
     @Test
     void shouldNotDisplaySeriesPosition_whenBookDoesNotHaveSeriesPosition() {
         // given
-        bookForm = createBookForm(false);
+        bookForm = createBookForm(READ, false);
 
         bookForm.openForm();
 
         // then
         Assertions.assertFalse(bookForm.seriesPositionFormItem.isVisible());
     }
 
+    /**
+     * Tests whether a book is saved to the database when the save-event is called
+     */
+    @Test
+    void shouldSaveBookToDatabase() {
+        // given
+        bookForm = createBookForm(TO_READ, false);
+
+        // when
+        bookForm.addListener(BookForm.SaveEvent.class, event -> bookService.save(event.getBook()));
+        bookForm.saveButton.click();
+
+        // then
+        Assertions.assertEquals(1, bookService.count());
+        Book bookInDatabase = bookService.findAll().get(0);
+        assertBookInSpecificShelf(TO_READ, bookInDatabase);
+    }
+
+    /**
+     * Tests whether a book is updated in the database when the save-event is called
+     */
+    @Test
+    void shouldUpdateValuesForBook() {
+        // given
+        bookForm = createBookForm(TO_READ, false);
+        bookForm.addListener(BookForm.SaveEvent.class, event -> bookService.save(event.getBook()));
+        bookForm.saveButton.click();
+
+        Book savedBook = bookService.findAll().get(0);
+        Assertions.assertEquals(bookTitle, savedBook.getTitle());
+        Assertions.assertEquals(TO_READ, savedBook.getPredefinedShelf().getPredefinedShelfName());
+        Assertions.assertEquals(firstName, savedBook.getAuthor().getFirstName());
+        Assertions.assertEquals(lastName, savedBook.getAuthor().getLastName());
+        Assertions.assertEquals(genre, savedBook.getGenre());
+        Assertions.assertNull(savedBook.getDateStartedReading());
+        Assertions.assertNull(savedBook.getDateFinishedReading());
```

Comment:
Do we need these assertions in this test? I guess the test is testing if a book can be updated so not sure if we want to assert a book loaded from DB (anyways we have a different test for that just above this test and if want to assert we might want to do in the above test?)

Reply:
They are not necessarily needed, I just added them to make clear, that the values are different after updating.
But you are right, we do save the same book in the test before and the values fit there, so I will remove these assertions. That also makes the test more readable.